lib/mix/lib/mix/tasks/archive.install.ex

```diff
@@ -104,6 +105,21 @@ defmodule Mix.Tasks.Archive.Install do
 
   ### Private helpers
 
+  defp resolve_destination(ez_path, contents) do
+    case :zip.list_dir(contents) do
+      {:ok, [_comment, zip_first_file | _]} ->
+        zip_root_dir =
+          zip_first_file
```

Comment:
Why did you remove the pattern matching? I don't think @lexmag suggested to remove it, rather to use `=` instead of `<-`.